packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,157 @@
+import { PriceFeedInterface } from "./PriceFeedInterface";
+import Web3 from "web3";
+import { BN } from "../types";
+import type { Logger } from "winston";
+import { InsuredBridgeL1Client, Relay } from "../clients/InsuredBridgeL1Client";
+import { InsuredBridgeL2Client, Deposit } from "../clients/InsuredBridgeL2Client";
+
+const { toBN, toWei } = Web3.utils;
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond True or False
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly convertPrice: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
```

Comment:
why did you make this an object? is this not somewhat non-standard with other price feeds?

Reply:
I think it might be better to follow the pattern done in other places for this kind of constructor. it will also mean you can use the `readonly` syntax which means you wont have to define `this.x` instance of each variable.

Reply:
I agree in general about consistency but other pricefeeds like FundingRateMultiplierPriceFeed and VaultPriceFeed also use this constructor logic and I think in general that using an object is better. WDYT?

Reply:
I think either way is fine. Only thing that's a little annoying about using an object is that you end up declaring the Params type and you can't do stuff like this
```ts
constructor(
  readonly private logger: Logger,
  readonly private web3: Web3,
  ...
)
```
to automatically declare instance variables in-line so you don't need to do:
```ts
private readonly logger: Logger;
private readonly web3: Web3;

....
this.logger = logger;
this.web3 = web3;
```

So basically the other way is just a bit more concise in ts, but either way is fine, and I can still see arguments for doing it this way 🤷 .

Reply:
I'm going to keep it as an object since it seems easier for clients and we should backlog a task to update all pricefeed constructors to use objects.